### ISSUE TITLE
Default docker-compose broken

### ISSUE THREAD

**Issue by NotUnlikeTheWaves** [Southclaws/cj#188]
After copying example.env to .env and filling in all channel related info and trying to run it, CJ cannot connect to the database at all as it cannot find a server.

It's 2am so debugging goes for another time, but maybe someone has some thoughts on it?

```
$ docker-compose up
WARNING: The DATA_DIR variable is not set. Defaulting to a blank string.
Recreating cj_mongodb_1 ... done
Recreating cj_cj_1      ... done
Attaching to cj_mongodb_1, cj_cj_1
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] MongoDB starting : pid=1 port=27017 dbpath=/data/db 64-bit host=1d0163dab3e2
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] db version v3.4.24
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] git version: 865b4f6a96d0f5425e39a18337105f33e8db504d
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.0.2g  1 Mar 2016
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] allocator: tcmalloc
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] modules: none
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] build environment:
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten]     distmod: ubuntu1604
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten]     distarch: x86_64
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten]     target_arch: x86_64
mongodb_1  | 2020-11-26T01:16:43.827+0000 I CONTROL  [initandlisten] options: { security: { authorization: "enabled" } }
mongodb_1  | 2020-11-26T01:16:43.830+0000 I -        [initandlisten] Detected data files in /data/db created by the 'wiredTiger' storage engine, so setting the active storage engine to 'wiredTiger'.
mongodb_1  | 2020-11-26T01:16:43.830+0000 I STORAGE  [initandlisten] 
mongodb_1  | 2020-11-26T01:16:43.830+0000 I STORAGE  [initandlisten] ** WARNING: Using the XFS filesystem is strongly recommended with the WiredTiger storage engine
mongodb_1  | 2020-11-26T01:16:43.830+0000 I STORAGE  [initandlisten] **          See http://dochub.mongodb.org/core/prodnotes-filesystem
mongodb_1  | 2020-11-26T01:16:43.830+0000 I STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=15549M,session_max=20000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000),checkpoint=(wait=60,log_size=2GB),statistics_log=(wait=0),verbose=(recovery_progress),
mongodb_1  | 2020-11-26T01:16:43.932+0000 I STORAGE  [initandlisten] WiredTiger message [1606353403:932350][1:0x7f4a4b936d40], txn-recover: Main recovery loop: starting at 20/4096
mongodb_1  | 2020-11-26T01:16:43.999+0000 I STORAGE  [initandlisten] WiredTiger message [1606353403:999848][1:0x7f4a4b936d40], txn-recover: Recovering log 20 through 21
mongodb_1  | 2020-11-26T01:16:44.038+0000 I STORAGE  [initandlisten] WiredTiger message [1606353404:38357][1:0x7f4a4b936d40], txn-recover: Recovering log 21 through 21
mongodb_1  | 2020-11-26T01:16:44.089+0000 I FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/data/db/diagnostic.data'
mongodb_1  | 2020-11-26T01:16:44.089+0000 I NETWORK  [thread1] waiting for connections on port 27017
cj_1       | {"level":"fatal","ts":"2020-11-26T01:16:55.792Z","caller":"bot/core.go:58","msg":"failed to connect to database","error":"no reachable servers","stacktrace":"github.com/Southclaws/cj/bot.Start\n\t/cj/bot/core.go:58\nmain.main\n\t/cj/main.go:26\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:200"}
cj_1       | {"level":"fatal","ts":"2020-11-26T01:17:08.124Z","caller":"bot/core.go:58","msg":"failed to connect to database","error":"no reachable servers","stacktrace":"github.com/Southclaws/cj/bot.Start\n\t/cj/bot/core.go:58\nmain.main\n\t/cj/main.go:26\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:200"}
cj_cj_1 exited with code 1
cj_1       | {"level":"fatal","ts":"2020-11-26T01:17:20.490Z","caller":"bot/core.go:58","msg":"failed to connect to database","error":"no reachable servers","stacktrace":"github.com/Southclaws/cj/bot.Start\n\t/cj/bot/core.go:58\nmain.main\n\t/cj/main.go:26\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:200"}

```

**Comment by NotUnlikeTheWaves** [comment]
Also to follow up on this, if I expose the mongodb on the default (public) network and try to connect to it I get `mongodb_1  | 2020-11-26T01:12:21.048+0000 I ACCESS   [conn2] SCRAM-SHA-1 authentication failed for root on cj from client 172.21.0.1:58144 ; UserNotFound: Could not find user root@cj` which I also can't explain why this is happening

**Comment by NotUnlikeTheWaves** [comment]
Fixed in #189
